Align noTraceSuppression val names to system property name

janekdb opened the pull request:
For vals based on the scala.control.noTraceSuppression system property
align the val name to the property name.

janekdb commented:
Hi,

Review by @adriaanm @retronym @SethTisue .

I've submitted this PR to get some advice on how to make this change conform with any requirements on deprecation or backwards compatibility. What are the additional or different changes that should be made on this commit?

Thanks,
Janek

adriaanm commented:
Thanks, @janekdb! Our compatibility policy requires deprecating a public member before we can rename it. One way is to introduce the correctly spelled one, leaving the old one as a deprecated alias:

``` diff
-  lazy val noTraceSupression   = bool("scala.control.noTraceSuppression", "scala should not suppress any stack trace creation")
+ @deprecated(...)
+  def noTraceSupression        = noTraceSuppression
+  lazy val noTraceSuppression  = bool("scala.control.noTraceSuppression", "scala should not suppress any stack trace creation")
```

The tests are failing because your branch is too far behind the 2.12.x branch. You could fix that by running `git pull --rebase https://github.com/scala/scala.git 2.12.x`

janekdb committed:
Align noTraceSuppression val names to system property name

For vals based on the scala.control.noTraceSuppression system property
align the val name to the property name.

Deprecate the existing val.

janekdb commented:
@adriaanm - I made some changes that hopefully gets this closer to an acceptable state.

som-snytt commented:
Just an idea, is it worth renaming it to `showStackTraces` or similar? I showed `NoStackTrace` to a colleague, and the double negative induces a little hiccup. That's just a thought, since you're incurring a name change anyway. Also, paulp and adriaanm always beat the double negation drum, so I figured I'd have a go.

janekdb commented:
@paulp, @adriaanm - any preference?

retronym commented:
LGTM. This is niche enough that I'd rather not change the name beyond fixing the spelling error in the code the refers to the system property.

som-snytt commented:
It would've been cool if a grammar issue was what brought paulp back into the fold, just to comment. I'm going to find an issue where I can casually impugn the Scrabble dictionary.

retronym commented:
He's probably too busy sesquioxidizing to find time to comment here.

som-snytt commented:
Rust never sleeps.

paulp commented:
I'm not psychologically strong enough to face up to having been responsible for a spelling error.
